internal/envoy/middleware.go

```diff
@@ -49,30 +50,20 @@ func GatewayFromContext(ctx context.Context) core.GatewayID {
 	return value.(core.GatewayID)
 }
 
-// GatewaySecretRegistry is used as the authority for determining what gateways the SDS server
-// should actually respond to because they're managed by consul-api-gateway
-type GatewaySecretRegistry interface {
```

Comment:
This abstraction gets dropped because we just use the store directly and check via the gateway itself whether or not it references the given secrets